zstd-decompress.c

```diff
@@ -66,37 +49,59 @@ int zstd_decompress_file(const char *filename)
 	}
 	close(input_file_fd);
 
+	ZSTD_DCtx *zstd_context = ZSTD_createDCtx();
```

Comment:
If you don't want to keep the context around, you can just use the non-context versions of `ZSTD_decompressDCtx()`. But what are you trying to achieve with this? The commit message doesn't have any rationale.

Reply:
I used the context variant because the documentation made it sound like it made sense for this use-case as we would be doing multiple decompressions, so I wrote a comment. @lumag then indicated he would like to see this use it: https://github.com/linux-msm/tqftpserv/pull/20#discussion_r1626213798. However, I didn't do any testing to investigate how this actually would affect performance and like I said I just blindly went off what the docs said. If others prefer not using the context approach, I'm fine with changing it.

Reply:
Granted that tqftpserv handles only few files during the whole lifetime and most of devices don't have swap, I think it's fine to use non-context versions of the functions. Let's free the memory for other software.